### PR TITLE
Feature/schiltz3/db migrate

### LINKED ISSUE #33: Add Channel integration to CSSC-Bot
Check for channels that match cs roles in the cs category and then create those channels if they don't exist. Save the channel name and ID to the class schema so it can be referenced easily later

### LINKED ISSUE #35: Standardize database role schema

### LINKED ISSUE #49: Use TITLE and INFO in channel topic
Channels are hard to identify because the channel topic does not have the class title in it

### LINKED ISSUE #50: Put TITLE in duplicate channel and role names
Instead of using \<name\>-0, \<name\>-1 to differentiate between duplicate names, use \<name\>-\<title1\> and \<name\>-\<title2\>

### PR DESCRIPTION
**Pull request by schiltz3**
# Migrate database to clean up structure & rewrite database calls to match

Edit the structure

* CODE => NAME
* TITLE
* INFO
* ~~ROLE_NAME~~
* ~~CHANNEL_NAME~~
* ROLE_ID
* CHANNEL_ID
* ~~UUID~~
* \+ DUPE
* \+ ACTIVE

This amounts to:
* Removing `UUID`, `ROLE_NAME` and `CHANNEL_NAME`.
* Cleaning up `CODE` and replacing it with `NAME`.
* Adding the `DUPE` field to signify if a class is a duplicate.
* Adding `ACTIVE` to specify if the course is being offered that semester. (Currently, all classes are considered active.)

### Steps to migrate database and discord server
Run `migrateDb` command to migrate the database
Run `CreateChannels` command to update channel names from the database
Run `CreateRoles` command to update role names from the database

# Link to github cards
#33 
#35 
#49 
#50

# Additions

* Add a helper function to create embeds
* Add `migrateDB` command to update the database to the new schema, update how duplicate codes are handled, and clean all strings.
* Add modified JS Docs to function. (Types are not added since this is specified in function definitions with typescript)

# Changes

* Change db scheme
* Change role and channel db calls to reflect changes
* Change how duplicate classes are handled
* Use course title and info in channel topics

## Bugfix
* replace all contiguous whitespace with a dash instead of replacing spaces with a dash
* Replace contiguous dashes with a dash in ROLE_NAME and CHANNEL_NAME
* Trim leading and trailing whitespace from TITLE
* Limit role name and channel name to 100 characters each
* Limit channel topic to 1024 characters each
* Only reply to role selection if role is actually assigned
* Add bottleneck package to limit the bot to 50 API calls per second
* Prefers matching on channel ID's rather than names

### CODE REVIEW

File: src/utils/embeds.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @return {*} - MessageEmbed
+ */
+export function create_default_embed(
+  client: Client,
+  title: string,
+  description: string
+) {
+  const color = "#0099ff";
+  const thumbnail =
+    "https://playantares.com/resources/CSSC-bot/cssc-server-icon.png";
```
Comment (nathen418) [inline src/utils/embeds.ts]: domain should be changed to `antaresnetwork.com`

Comment (schiltz3) [reply src/utils/embeds.ts]: for some reason the change from staging didn't come through, so I fixed all 13 occureneces

File: src/utils/roles.ts [hunk 2]
```diff
@@ -63,18 +72,10 @@ export async function addNewRole<T extends IRole>(
...
-  let role;
-  switch (model.modelName) {
-    case "class":
-      //TODO: Rewrite IRole to include name, and replace CODE with name
-      role = await model.findOne({ CODE: id });
-      break;
-    default:
-      role = await model.findOne({ NAME: id });
-  }
+  const role = await model.findOne({ ROLE_NAME: id });
```
Comment (nathen418) [inline src/utils/roles.ts]: `ROLE_NAME` should be changed to `NAME`
`NAME` is the pk in the database used to match the interaction.option to the db entry. maybe the pk name should be changed to `ID` or something to reduce confusion.

Comment (schiltz3) [reply src/utils/roles.ts]: I agree. I swapped the variable in the function to be name, then added a note in documentation, we can open an issue to swap it to name or code

**Review by nathen418 — APPROVED** [review]
LGTM

**Pull request merged**